presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PruneSemiJoinColumns.java

```diff
@@ -47,12 +48,17 @@ public PruneSemiJoinColumns()
                 referencedOutputs.stream()
                         .filter(symbol -> !symbol.equals(semiJoinNode.getSemiJoinOutput())),
                 Stream.of(semiJoinNode.getSourceJoinSymbol()),
-                semiJoinNode.getSourceHashSymbol().map(Stream::of).orElse(Stream.empty()))
+                semiJoinNode.getSourceHashVariable().map(this::toSymbol).map(Stream::of).orElse(Stream.empty()))
                 .collect(toImmutableSet());
 
         return restrictOutputs(idAllocator, semiJoinNode.getSource(), requiredSourceInputs)
                 .map(newSource ->
                         semiJoinNode.replaceChildren(ImmutableList.of(
                                 newSource, semiJoinNode.getFilteringSource())));
     }
+
+    private Symbol toSymbol(VariableReferenceExpression variable)
```

Comment:
Do we wanna make this a generic util or just remove this function? `new Symbol(var.getName())` has been used widely in the previous commits.

Reply:
I think I just got lazy for once and added this. By the end of this refactoring this would not exist anyways. So just pretend they are not here 😛